File: src/main/java/io/github/dqualizer/dqlang/types/adapter/symbolics/SymbolicLoad.java
package io.github.dqualizer.dqlang.types.adapter.symbolics;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.adapter.symbolics.generic.SymbolicLoadType;
import lombok.Getter;
import lombok.ToString;

import java.util.concurrent.TimeUnit;

@Getter
@ToString
public class SymbolicLoad {

    private SymbolicLoadType<Integer> integer;
    private SymbolicLoadType<Double> decimal;

    /**
     * TimeUnit for open loads,
     * @example users per SECONDS, users per MINUTES
     */
    @JsonProperty("time_unit")
    private TimeUnit timeUnit;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/symbolics/SymbolicLoad.kt
package io.github.dqualizer.dqlang.types.adapter.symbolics

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.adapter.symbolics.generic.SymbolicLoadType
import java.util.concurrent.TimeUnit

data class SymbolicLoad(
  val integer: SymbolicLoadType<Int>,

  val decimal: SymbolicLoadType<Double>,

  /**
   * TimeUnit for open loads,
   * @example users per SECONDS, users per MINUTES
   */
  @JsonProperty("time_unit")
  val timeUnit: TimeUnit
)
